### PR TITLE
slack: init aarch64 darwin

### PR DESCRIPTION
**Pull request by hlolli**
###### Motivation for this change

Missing support

###### Things done

I had to remove the "defaults write com.tinyspeck.slackmacgap" part from aarch64, otherwise `2021-05-31 08:58:32.751 defaults[36345:252092] Could not write domain com.tinyspeck.slackmacgap; exiting`

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/slack/default.nix [hunk 1]
```diff
@@ -174,7 +182,8 @@ let
     installPhase = ''
       mkdir -p $out/Applications/Slack.app
       cp -R . $out/Applications/Slack.app
-      /usr/bin/defaults write com.tinyspeck.slackmacgap SlackNoAutoUpdates -bool YES
+    '' + lib.optionalString (!stdenv.isAarch64) ''
```
Comment (prusnak) [inline pkgs/applications/networking/instant-messengers/slack/default.nix:185]: Why this change?

Comment (hlolli) [reply pkgs/applications/networking/instant-messengers/slack/default.nix:185]: as I wrote in "things done"

I had to remove the "defaults write com.tinyspeck.slackmacgap" part from aarch64, otherwise 2021-05-31 08:58:32.751 defaults[36345:252092] Could not write domain com.tinyspeck.slackmacgap; exiting

Comment (prusnak) [reply pkgs/applications/networking/instant-messengers/slack/default.nix:185]: Please put the explanation in the comment why this is needed

Comment (hlolli) [reply pkgs/applications/networking/instant-messengers/slack/default.nix:185]: finally, done (or could I make the comment more explanitory?)

Comment (prusnak) [reply pkgs/applications/networking/instant-messengers/slack/default.nix:185]: I think that's OK

**Comment by ethancedwards8** [comment]
Can you fix the merge conflicts and squash the commits?

**Comment by siraben** [comment]
Please rebase.

**Comment by hlolli** [comment]
I didn't see that some aarch64 support was there. But it was missing some code which I feel justifies this PR. I built, ran and tested that Slack runs with the version already declared with these changes merged.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 125051` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>slack</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 125051` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>slack</li>
  </ul>
</details>

File: pkgs/applications/networking/instant-messengers/slack/default.nix [hunk 2]
```diff
@@ -56,28 +56,27 @@ let
...
+
+  src = let
+    base = "https://downloads.slack-edge.com";
+  in {
+    x86_64-darwin = fetchurl {
+      url = "${base}/releases/macos/${version}/prod/x64/Slack-${version}-macOS.dmg";
+      sha256 = x86_64-darwin-sha256;
+    };
+    x86_64-linux = fetchurl {
+      url = "${base}/linux_releases/slack-desktop-${version}-amd64.deb";
```
Comment (kirillrdy) [inline pkgs/applications/networking/instant-messengers/slack/default.nix:72]: i think url is wrong, after this PR slack on linux can no longer be fetched

Comment (kirillrdy) [reply pkgs/applications/networking/instant-messengers/slack/default.nix:72]:
```
"${base}/releases/linux/${version}/prod/x64/slack-desktop-${version}-amd64.deb"
```
this works for me on linux

File: pkgs/applications/networking/instant-messengers/slack/default.nix [hunk 3]
```diff
@@ -195,7 +194,10 @@ let
       runHook preInstall
       mkdir -p $out/Applications/Slack.app
       cp -R . $out/Applications/Slack.app
-      /usr/bin/defaults write com.tinyspeck.slackmacgap SlackNoAutoUpdates -bool YES
+    '' + lib.optionalString (!stdenv.isAarch64) ''
+      # on aarch64-darwin we get: Could not write domain com.tinyspeck.slackmacgap; exiting
+      /usr/bin/defaults write com.tinyspeck.slackmacgap SlackNoAutoUpdates -Bool YES
```
Comment (spinylobster) [inline pkgs/applications/networking/instant-messengers/slack/default.nix:199]: https://github.com/hlolli/nixpkgs/blob/5facd430d4aedae75f8c4027fa62624225b9e752/pkgs/applications/networking/instant-messengers/slack/default.nix#L199
`-Bool` is wrong. `-bool` was correct.
When Intel Mac users are trying install Slack, it fails.

Comment (alex-demetriou) [reply pkgs/applications/networking/instant-messengers/slack/default.nix:199]: > `-Bool` is wrong. `-bool` was correct.

Yup, I can run the command myself successfully when I use the correct casing `-bool`.
From the `installPhase`, though, it's still not happy.
> When Intel Mac users are trying install Slack, it fails.

Looks like the original issue was not limited to `aarch64-darwin` after all. Maybe something changed at the OS level. I'm having the same issue on `x86_64`.

Comment (tylerlevine) [reply pkgs/applications/networking/instant-messengers/slack/default.nix:199]: Same thing here, running `/usr/bin/defaults write com.tinyspeck.slackmacgap SlackNoAutoUpdates -bool YES` works in a terminal, but fails with `defaults[2342:1734766] Could not write domain com.tinyspeck.slackmacgap; exiting` when run from the `installPhase`.
